formats/about/SPEC_1.0.md

```diff
@@ -0,0 +1,61 @@
+# About format
+
+<dl>
+	<dt>Namespace</dt><dd>`about`</dd>
+	<dt>Version</dt><dd>1.0</dd>
+</dl>
+
+This is an application format for describing personal information (e.g. display names, avatars, bios) associated with particular author keypairs.
+
+## Keypair association
+
+Personal information should only be publishable and modifiable by the holder of a specific keypair. 
+
+Therefore, each documents following this format is prefixed with the following:
+
+`/about/1.0/~{KEYPAIR_ADDRESS}`
+
+Where `KEYPAIR_ADDRESS` is the full public address of the keypair associated with this document, e.g. `@suzy.bo5sotcncvkr7p4c3lnexxpb4hjqi5tcxcov5b4irbnnz2teoifua`.
+
+The presence of an author keypair prefixed with a tilde in a path invokes path ownership, and only this keypair will be able to write to these paths.
+
+## Display names
+
+Author keypairs begin with a shortname to make the keypair easier to identify, e.g. `suzy`. These shortnames cannot be changed and are always four characters long. This makes them unsuitable for most names and the fluctuating nature of identity.
+
+Therefore it is desirable to have a free-form display name which can be changed at any time, and which can differ between each share.
+
+### Display name document path
+
+The path format of a display name document is:
+
+```
+`/about/1.0/~{KEYPAIR_ADDRESS}/displayName
+```
+
+### Display name document fields
+
+The `text` property of the document is the display name the keypair's owner wishes to use. It can be any UTF-8 string.
+
+## Avatars
+
+An image can be associated with an author keypair.
+
+### Avatar document path
+
+The path format of a display name document is:
+
+```
+`/about/1.0/~{KEYPAIR_ADDRESS}`/avatar.{IMAGE_EXTENSION}
+```
+
+where `IMAGE_EXTENSION` is either `gif`, `jpg`, or `png`, corresponding to the format of the data used with this document's attachment.
```

Comment:
Only one such document may be present, perhaps? Also, is a client supposed to try all three in order?

Reply:
Oh, right. This doesn't stop someone from setting all three.

For simplicity's sake, I'd love for clients to be able to generate a path for the avatar pretty deterministically. 

On the other hand, I didn't want to wade into "let's choose one image format for all time". Then either users have to convert image formats, or this conversion needs to be done by clients.

Maybe falling through the different extensions is the best. What do you think?

Reply:
I'd lean towards not limiting the image format here if possible for a couple of reasons:

1. Image formats can change and improve. Don't have any specific examples but there's a chance that newly introduced formats can be preferable, depending on the implementers context.

2. It feels like more a concern for the client to decide and worry about the image format that's being used. This way there's more flexibility for them to design their application as they see fit. As long as there's a path that matches `avatar.*`, that should be enough.